src/modules/dashboard/components/Dialogs/ControlSafeDialog/ControlSafeDialog.tsx

```diff
@@ -86,6 +91,14 @@ const ControlSafeDialog = ({
     expandedValidationSchema,
   );
 
+  const handleSelectedContractMethods = (
+    updatedMethods: UpdatedMethods,
+    validateForm: FormikHelpers<FormValues>['validateForm'],
+  ) => {
+    setSelectedContractMethods(updatedMethods);
+    setTimeout(validateForm, 0);
+  };
```

Comment:
The things we do to disable a button

Reply:
@willm30 Can I just ask why you are using `setTimeout` here? Not saying it's wrong, I'm just curious

Reply:
@jakubcolony It's to avoid a problem in formik where the form validates against state that is incorrect/not uptodate. If you take out the setTimeout, the errors object will be incorrect. For example, I've noticed that even if a safe is selected, form validation will produce an error saying that the safe should not be null (when it isn't). It's a strange one for sure, and may have something to do with https://github.com/jaredpalmer/formik/issues/529.